modules/seqexec/server/src/main/scala/seqexec/server/tcs/Gaos.scala

```diff
@@ -28,45 +30,125 @@ trait Gaos[F[_]] {
 }
 
 object Gaos {
-  sealed trait PauseCondition
-
-  // Telescope offset will be changed
-  final case class OffsetMove(previousOffset: FocalPlaneOffset, newOffset: FocalPlaneOffset) extends PauseCondition
-  // OI will be turn off
-  case object OiOff extends PauseCondition
-  // PI will be turn off
-  case object P1Off extends PauseCondition
-  // Unguided step
-  case object GaosGuideOff extends PauseCondition
-
-  sealed trait ResumeCondition
-  // Telescope offset will be changed
-  final case class OffsetReached(newOffset: FocalPlaneOffset) extends ResumeCondition
-  // OI will be turn off
-  case object OiOn extends ResumeCondition
-  // PI will be turn off
-  case object P1On extends ResumeCondition
-  // Guided step
-  case object GaosGuideOn extends ResumeCondition
-
-  implicit val becauseOffsetMoveEq: Eq[OffsetMove] = Eq.by(x => (x.previousOffset, x.newOffset))
-
-  implicit val pauseReasonEq: Eq[PauseCondition] = Eq.instance{
-    case (a: OffsetMove, b: OffsetMove) => a === b
-    case (OiOff, OiOff)                 => true
-    case (P1Off, P1Off)                 => true
-    case (GaosGuideOff, GaosGuideOff)   => true
-    case _                              => false
+  sealed trait PauseCondition extends Product with Serializable
```

Comment:
I think the `PauseConditionSet` is a great idea, but it sort of bugs me that `PauseCondition` and `ResumeCondition`, along with `PauseConditionSet` and `ResumeConditionSet` are completely isomorphic and yet require all that duplicated code.  At the same time, having a distinct type for pause set vs. resume set is valuable. Maybe combine them in one `Condition` and `ConditionSet` and then wrap the set to give them distinct types?

Reply:
I see your point, but honestly, I don't thing the gain is worth my time.